### PR TITLE
So much changes

### PR DESCRIPTION
**Pull request by jeandudey**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
 CMakeFiles
 Makefile
 *.a
-*.cmake
```
Comment (kahowell) [inline .gitignore:4]: This is a rule I generally put for the case of "in-source" builds, so that my build files aren't picked up if I lazily build in the same directory.

File: CMake/Modules/FindPhysicsFS.cmake [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#.rst:
```
Comment (kahowell) [inline CMake/Modules/FindPhysicsFS.cmake]: Why duplicate the FindPhysicsFS.cmake module from CMake? It looks like you're trying to do some things specific to your setup; I'd instead recommend using facilities built in to the vanilla CMake module. Take a look at https://github.com/Kitware/CMake/blob/master/Modules/FindPhysFS.cmake. Looks like you should be able to set `PHYSFSDIR` environment variable.

File: README.md [hunk 3]
```diff
@@ -19,3 +25,7 @@ PhysFS::Util namespace.
  - Both ifstream and ofstream are standard streams, and only have a couple of
 extra methods - `eof` and `length`, which call `PHYSFS_eof` and 
 `PHYSFS_fileLength` respectively.
+
```
Comment (kahowell) [inline README.md]: I like all other changes in this file, except for build section, because building really depends on how you're trying to integrate the lib. If you rephrased to give this as an example, and maybe point to general CMake build directions, that'd be much better.

File: include/physfs.hpp [hunk 4]
```diff
@@ -8,171 +8,116 @@
...
+    std::string utf8ToUcs4(char const * src);
+    std::string utf8FromUcs2(uint16 const * src);
+    std::string utf8ToUcs2(char const * src);
+    std::string utf8FromLatin1(char const * src);
 }
 
 }
 
-#endif /* _INCLUDE_PHYSFS_HPP_ */
+#endif /* _INCLUDE_PHYSFS_HPP_ */
```
Comment (kahowell) [inline include/physfs.hpp]: I like these changes, and will include probably include them verbatim, once other issues are addressed (or if you did a pull request with just these).

File: src/physfs.cpp [hunk 5]
```diff
@@ -103,289 +105,304 @@ base_fstream::base_fstream(PHYSFS_File* file) : file(file) {
...
+std::string Util::utf8FromLatin1(const char* src) {
+    std::string value;
+    std::size_t length = strlen((char*) src) * 2;
+    char * buffer = new char[length]; // will be smaller than len
+    PHYSFS_utf8FromLatin1(src, buffer, length);
+    value.append(buffer);
+    return value;
 }
 
 }
```
Comment (kahowell) [inline src/physfs.cpp:391]: Similar to my comment about the header file, I like these changes, and will include them verbatim, if you address other issues with this pull request or create a separate pull request.

**Comment by kahowell** [comment]
A lot of changes for a single pull request. I would like to request that you split this out into several pull requests. Logically, I see:
- formatting changes
- README updates
- build changes

In general I am cool with the formatting changes and README updates (one small caveat w/ the README). Build changes on the other hand, may need more discussion.

If you don't want to do more pull requests, just let me know, and I'll cherry-pick some of your changes in.

Thanks for sharing :-)

File: CMake/Modules/FindPhysicsFS.cmake (hunk 2 above)
Comment (jeandudey) [inline CMake/Modules/FindPhysicsFS.cmake]: I modified the original that comes with CMake but i never tried to use the PHYSFSDIR environment variable. Thanks.
